homeassistant/components/datadog/config_flow.py

```diff
@@ -0,0 +1,101 @@
+"""Config flow for Datadog."""
+
+from typing import Any
+
+from datadog import DogStatsd
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult, OptionsFlow
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_PREFIX
+from homeassistant.core import HomeAssistant, callback
+
+from .const import (
+    CONF_RATE,
+    DEFAULT_HOST,
+    DEFAULT_PORT,
+    DEFAULT_PREFIX,
+    DEFAULT_RATE,
+    DOMAIN,
+)
+
+
+class DatadogConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Datadog."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle user config flow."""
+        errors: dict[str, str] = {}
+        if user_input:
+            # Validate connection to Datadog Agent
+            success = await validate_datadog_connection(
+                self.hass,
+                DogStatsd(user_input["host"], user_input["port"], user_input["prefix"]),
+            )
+            if not success:
+                errors["base"] = "cannot_connect"
+            else:
+                return self.async_create_entry(
+                    title="Datadog", data={}, options=user_input
```

Comment:
```suggestion
                    title=f"Datadog {user_input["host"]}", data={}, options=user_input
```
Since you can have many maybe append the host to the config entry title.